prov/usnic/src/usdf_cm.c

```diff
@@ -563,26 +580,85 @@ usdf_cm_msg_connect(struct fid_ep *fep, const void *addr,
 	return ret;
 }
 
+/* A wrapper to core function to translate string address to
+ * sockaddr_in type. We are expecting a NULL sockaddr_in**.
+ * The core function will allocated it for us. The caller HAS TO FREE it.
+ */
+int usdf_str_toaddr(const char *str, struct sockaddr_in **outaddr)
+{
+	uint32_t type;
+	size_t size;
+	int ret;
+	char outstr[USDF_ADDR_STR_LEN];
+
+	type = FI_SOCKADDR_IN;
+
+	/* call the core function. The core always allocate the addr for us. */
+	ret = ofi_str_toaddr(str, &type, (void **)outaddr, &size);
+
+	/* verify the output in debug. */
+	inet_ntop(AF_INET, &((*outaddr)->sin_addr), outstr, size);
```

Comment:
Hmm, this is always going to happen. Maybe we want to put this inside an `#if DEBUG` block or something.